src/site/topics/trees/linked-binary-search-trees.rst

```diff
@@ -2,473 +2,290 @@
-
-* Here we have two constructors
-    * A default constructor, which just sets up an empty binary search tree
-    * A constructor that we could give an element to be the tree's root
-        * This constructor is probably unnecessary, but let's keep it
 
 
 Static Node Class
 =================
 
-* You likely noticed the use of ``Node<T> root``
-* Like the linked stack and linked queue, we will make use of nodes
-* You may remember that we had two options for doing this
-    1. Simply create a top level node class
-    2. Create a :doc:`nested node class aside </topics/stacks/nested>`
-
-* To keep things clean, we will go with option #2
-
-.. code-block:: java
-    :linenos:
-
-    // This class exists within the LinkedBinarySearchTree class
-    private static class Node<T> {
-
-        private T data;
-        private Node<T> left;
-        private Node<T> right;
-
-        private Node(T data) {
-            this.data = data;
-            this.left = null;
-            this.right = null;
-        }
-
-        private T getData() {
-            return data;
-        }
-
-        private void setData(T data) {
-            this.data = data;
-        }
-
-        private Node<T> getLeft() {
-            return left;
-        }
-
-        private void setLeft(Node<T> left) {
-            this.left = left;
-        }
-
-        private Node<T> getRight() {
-            return right;
-        }
-
-        private void setRight(Node<T> right) {
-            this.right = right;
-        }
-    }
+* Notice the use of ``Node<T> root``
+* Like the linked stack and linked queue, this linked binary search tree will make use of nodes
+* To keep things clean, a nested node class is used
 
+.. literalinclude:: /../main/java/LinkedBinarySearchTree.java
+    :language: java
+    :lineno-match:
+    :lines: 447-491
 
 * Things of note for this node class
+
     * This is a binary tree node, thus it has a left and right reference for the left and right subtrees
     * This nested static class is contained within the ``LinkedBinarySearchTree`` class
 
 
+
 Add to Binary Search Tree
 =========================
 
-.. code-block:: java
-    :linenos:
+.. literalinclude:: /../main/java/LinkedBinarySearchTree.java
+    :language: java
+    :lineno-match:
+    :lines: 15-40
 
-    public void add(T element) {
-        root = add(element, root);
-        size++;
-    }
 
-    private Node<T> add(T element, Node<T> current) {
-        if (current == null) {
-            return new Node<>(element);
-        } else if (current.getData().compareTo(element) > 0) {
-            current.setLeft(add(element, current.getLeft()));
-        } else {
-            current.setRight(add(element, current.getRight()));
-        }
-        return current;
-    }
-
-
-* We use a helper method to setup a call to the recursive method
+* A public helper method is used to setup a call to the private recursive add
```

Comment:
public methods aren't really helpers. This is more like, _the public method calls the private recursive add method with sane values_